README.md

```diff
@@ -27,9 +27,126 @@ const JsonML = require('jsonml.js/lib/utils');
 
 ## API
 
-### jsonml.js/lib/utils
+### DOM
+
+```js
+const { dom } = require('jsonml.js');
+```
+
+##### `fromHTML( Element: elem, Function: filter)`
+
+Create JsonML tree from DOM, given a root DOM Element.
+
+- `{Element}` elem - DOM [Element]
+- `{Function}` filter - optional method to apply on every created JsonML node
+
+[Element]: https://developer.mozilla.org/en-US/docs/Web/API/Element
+
+##### `fromHTMLText( string: html, Function: filter)`
+
+Create JsonML tree from HTML string.
+
+- `{string}` html - HTML string
+- `{Function}` filter - optional method to apply on every created JsonML node
+
+-----
+
+### HTML
+
+```js
+const { html } = require('jsonml.js');
+```
+
+#### `toHTML( jml, filter )`
+
+Generate an HTML DOM tree given a JsonML struture. An optional `filter` method can be passed and will be applied to each node.
```

Comment:
what should this `filter` function return?

Reply:
good point. i should document the signature.